### PR TITLE
Remove the global state for HotShot height

### PR DESCRIPTION
**Pull request by ImJeremyHe**
Remove the extra global state that we added for the hotshot block height. We don't need this anymore for following reasons:
- Since the L1 arbitrator and block validator have the ability to look back to the previous messages, we don't need to store the hotshot height in the global state for checking the continuity of hotshot block
- Removing the extra global state helps the nitro integration keep in line with the upstream so that we can utilize their tools, such as `arbitrum-sdk`, without any change.
- This change is also friendly to the migration

### This PR:
- Remove the extra global state.
- Remove the continuity check of hotshot block for now, this is not necessary in sovereign sequencer.
- Also remove the staker check which is largely based on this global state. We have to figure out how to do the similar things later.

### CODE REVIEW

**Review by Sneh1999 — APPROVED** [review]

**Review by Sneh1999 — COMMENTED** [review]
@ImJeremyHe I would suggest we merge [my](https://github.com/EspressoSystems/nitro-espresso-integration/pull/204) PR first. This would make the changes in this PR cleaner and the tests will also pass for this PR

File: cmd/replay/espresso_validation.go [hunk 1]
```diff
@@ -21,10 +22,9 @@ import (
 func handleEspressoPreConditions(message *arbostypes.MessageWithMetadata, isEnabled bool) (bool, func()) {
 	// calculate and cache all values needed to determine if the preconditions are met to enter the Espresso STF logic
 	isNonEspressoMessage := arbos.IsL2NonEspressoMsg(message.Message)
-	hotshotHeight := wavmio.GetEspressoHeight()
 
 	validatingEspressoLivenessFailure := isNonEspressoMessage && isEnabled
-	validatingEspressoHeightFailure := isNonEspressoMessage && hotshotHeight != 0
+	validatingEspressoHeightFailure := isNonEspressoMessage && isEnabled
```
Comment (sveitser) [inline cmd/replay/espresso_validation.go:27]: The two variables defined here the same now. Why keep both?

**Review by sveitser — APPROVED** [review]
LGTM but one question. I'm also fine with not addressing this right now because we can't panic due to an error in the Message in the replay binary anyway so we will probably have to make significant changes in that region anyway.

File: cmd/replay/espresso_validation.go (hunk 1 above)
Comment (ImJeremyHe) [reply cmd/replay/espresso_validation.go:27]: I am thinking we will refactor here. So I just want to make it work with minimum changes

Comment (sveitser) [reply cmd/replay/espresso_validation.go:27]: nit: not a refactor because we have to change functionality :laughing:

**Pull request merged**